src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentPresentation/UriPresentationParams.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using MediatR;
+using Newtonsoft.Json;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.DocumentPresentation
+{
+    /// <summary>
+    /// Class representing the parameters sent for a textDocument/_vs_textPresentation request.
+    /// </summary>
+    internal class UriPresentationParams : ITextDocumentIdentifierParams, IRequest<WorkspaceEdit?>, IBaseRequest
```

Comment:
If you rebase on `main` you should be able to utilize the actual LSP protocol types if you want instead of copying

Reply:
Err.. how? Or do you just mean for `WorkspaceEdit`? Because `UriPresentationParams` itself still needs to exist, because `VSInternalUriPresentationParams` doesn't implement `IRequest`

Reply:
... and all of our helpful extension methods don't work on VS types :(

Reply:
@davidwengier I created a "bridge" class in the completion case: https://github.com/dotnet/razor-tooling/commit/037233e1b22451c1b7b9297673067debeeba3385#diff-723eebfac1bc4590313e8e74914ec30fa238a3f0ca6101997dd561aad7f25498R15

Reply:
I would have thought that would have failed because of the different serialization attributes. Does it work for completion because the property names match the desired JSON property names? Because that isn't the case for VSInternal stuff. Or does it work because Newtonsoft honours DataMember attributes as well as its own?

I guess I'll try and find out ¯\\\_(ツ)_/¯

Reply:
Newtonsoft honors DataMember too 😄